src/coreclr/jit/lsra.cpp

```diff
@@ -3479,22 +3388,65 @@ regNumber LinearScan::allocateReg(Interval* currentInterval, RefPosition* refPos
             prevRegOptCandidates &= ~prevRegOptCandidateBit;
             regNumber prevRegOptCandidateRegNum = genRegNumFromMask(prevRegOptCandidateBit);
             Interval* assignedInterval          = physRegs[prevRegOptCandidateRegNum].assignedInterval;
-            // The assigned should be non-null, and should have a recentRefPosition, however since
-            // this is a heuristic, we don't want a fatal error, so we just assert (not noway_assert).
+            bool      foundPrevRegOptReg        = true;
```

Comment:
It looks to me like `foundPrevRegOptReg` will always be `true`; how can it be set to `false`?

Reply:
Yes, I went back and forth and seems I missed the case when it should be `false`. I have updated the code.

For reference, I was trying to add back the missing code that was present pre #45135,. You can see it at https://github.com/dotnet/runtime/pull/45135/files#diff-a985c84f41c0c758b9a2f849758cdfd6eac4acf6cfc2151a28d206d70e0358e0L3816-L3824 (need to load diffs for `lsra.cpp` changes).